lib/service/service.go

```diff
@@ -666,7 +666,11 @@ func NewTeleport(cfg *Config) (*TeleportProcess, error) {
 		} else {
 			switch cfg.JoinMethod {
 			case types.JoinMethodToken, types.JoinMethodUnspecified, types.JoinMethodIAM:
-				cfg.HostUUID = uuid.New().String()
+				rawID, err := uuid.NewRandom()
+				if err != nil {
+					return nil, trace.Wrap(err)
+				}
+				cfg.HostUUID = rawID.String()
```

Comment:
@codingllama @atburke Guys, we specifically wanted to avoid using NewString to avoid the panic and print a proper error message to users. This can (and does) happen when Teleport starts very early in the node initialization cycle and urandom isn't initialized yet. A customer is seeing this in their environment.

@atburke Can you please roll this back to the way it was originally implemented and make sure we return an proper error message as specified in https://github.com/gravitational/teleport/pull/11810#issuecomment-1093154334.

Reply:
Yeah, I think for now doing this only for node ID generation would suffice to address the customer issue, it's pretty "edge-casey" scenario for sure and only happens when Teleport starts for the very first time. Agree with adding a clarifying comment.